replication-api-impl/src/main/java/org/codice/ditto/replication/api/impl/data/ReplicatorConfigImpl.java

```diff
@@ -113,6 +119,7 @@ public void fromMap(Map<String, Object> properties) {
     setSuspended(Boolean.valueOf((String) properties.get(SUSPENDED_KEY)));
     setDeleted(Boolean.valueOf((String) properties.get(DELETED_KEY)));
     setDeleteData(Boolean.valueOf((String) properties.get(DELETE_DATA_KEY)));
+    setMetadataOnly(Boolean.valueOf((String) properties.get(METADATA_ONLY_KEY)));
```

Comment:
❗️ Looks to me like we should be checking the version we are getting and initializing the new properties with their default instead of blindly trying to extract values from a map. The current implementation can lead to differences between what is documented and what might happen

Reply:
These are assumptions (although most likely correct). I don't think we should rely on the presence or not of something to know which version it is; this is why we are saving a version number. We should rely on the version number and then not even look at the map when we know it shouldn't have been there to start with.

Reply:
added a version check